internal/app/fetcher/libraries.go

```diff
@@ -0,0 +1,77 @@
+package fetcher
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/xssnick/tonutils-go/tlb"
+	"github.com/xssnick/tonutils-go/tvm/cell"
+)
+
+type LibDescription struct {
+	_          tlb.Magic        `tlb:"$00"`
+	Lib        *cell.Cell       `tlb:"^"`
+	Publishers *cell.Dictionary `tlb:"dict inline 256"`
+}
+
+func (s *Service) GetAccountLibraries(ctx context.Context, raw *tlb.Account) (*cell.Cell, error) {
+	hashes, err := findLibraries(raw.Code)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "find libraries")
+	}
+
+	libs, err := s.API.GetLibraries(ctx, hashes...)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "get libraries")
+	}
+
+	libsMap := cell.NewDict(256)
+
+	for i, hash := range hashes {
+		desc := LibDescription{Lib: libs[i]}
+
+		h := cell.BeginCell().MustStoreSlice(hash, 256).EndCell()
+		t, err := tlb.ToCell(desc)
+
+		if err != nil {
+			return nil, err
+		}
+
+		err = libsMap.Set(h, t)
+		s.libraries.set(hash, &desc)
```

Comment:
That's something messy. We should check error here, not update cache. Cache should be updated right before the successful return of `GetAccountLibraries`.